### PR TITLE
Change format of metadata serialization according to spec

### PR DESCRIPTION
**Pull request by klokane**

### CODE REVIEW

File: src/refract/SerializeCompactVisitor.cc [hunk 1]
```diff
@@ -42,9 +42,17 @@ namespace refract
 
         typedef ArrayElement::ValueType::const_iterator iterator;
         for (iterator it = e.value.begin(); it != e.value.end(); ++it) {
-            SerializeCompactVisitor s;
-            (*it)->content(s);
-            array.push(s.value());
+            if ((*it)->renderType() == IElement::rFull) {
+                SerializeVisitor s;
+                s.visit(*(*it));
+                //(*it)->content(s);
```
Comment (pksunkara) [inline src/refract/SerializeCompactVisitor.cc]: What is with this line?

**Comment by pksunkara** [comment]
lgtm. Just one minor comment.

File: src/refract/SerializeCompactVisitor.cc (hunk 1 above)
Comment (klokane) [inline src/refract/SerializeCompactVisitor.cc]: sry forgotred. will fix

**Comment by klokane** [comment]
fixed

**Comment by pksunkara** [comment]
Oops. I didn't see that this PR is made to the array::create branch. I merged that PR already. Can you make a new PR?

PS: From next time, please mention if there are any dependent PRs.

**Comment by klokane** [comment]
#78

**Pull request closed without merging**